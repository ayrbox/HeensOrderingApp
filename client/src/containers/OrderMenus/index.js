import { connect } from 'react-redux';
import Menus from './OrderMenus';

// import { setMenu } from '../../actions/newOrderActions';

const mapState = state => ({
  orderType: state.newOrder.orderType,
  category: state.newOrder.categoryId,
});

export default connect(mapState, {
  setMenu: () => {},
})(Menus);
