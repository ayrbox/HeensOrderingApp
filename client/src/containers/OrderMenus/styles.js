const styles = {
  itemDesc: {
    width: '90%',
  },
};

export default styles;
