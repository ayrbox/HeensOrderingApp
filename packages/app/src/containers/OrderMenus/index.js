import { connect } from 'react-redux';
import Menus from './OrderMenus';

import { setMenu } from '../../store/newOrder';

const mapState = state => ({
  orderType: state.newOrder.orderType,
  category: state.newOrder.categoryId,
});

export default connect(mapState, {
  setMenu,
})(Menus);
