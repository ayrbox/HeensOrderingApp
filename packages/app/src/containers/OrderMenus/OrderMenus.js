import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';

import Fetch from '../../components/Fetch';
import styles from './styles';

const Menus = ({
  category,
  classes,
  setMenu,
}) => (
  <List
    cellHeight={160}
    className={classes.gridList}
    cols={3}
  >
    <Fetch url="/api/menus">
      {({ loading, data }) => {
        if (loading) {
          return 'Loading......';
        }
        return data
          .filter(({
            category: { _id: categoryId },
          }) => (category ? categoryId === category : true))
          .map((menu) => {
            const {
              _id: id,
              name,
              price,
              description,
            } = menu;
            return (
              <ListItem
                key={id}
                button
                alignItems="flex-start"
                onClick={(e) => {
                  e.preventDefault();
                  setMenu(menu);
                }}
              >
                <ListItemText
                  primary={<strong>{name}</strong>}
                  secondary={(
                    <Typography component="span" className={classes.itemDesc}>
                      {description}
                    </Typography>
                  )}
                />
                <ListItemSecondaryAction>
                  <Typography variant="h6">
                    {`Â£${price.toFixed(2)}`}
                  </Typography>
                </ListItemSecondaryAction>
              </ListItem>
            );
          });
      }}
    </Fetch>
  </List>
);

Menus.defaultProps = {
  category: undefined,
};

Menus.propTypes = {
  category: PropTypes.string,
  classes: PropTypes.shape().isRequired,
  setMenu: PropTypes.func.isRequired,
};

export default withStyles(styles)(Menus);
