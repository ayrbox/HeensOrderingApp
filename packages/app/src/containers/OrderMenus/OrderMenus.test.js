import React from 'react';
import {
  render,
  waitForElement,
  cleanup,
  fireEvent,
} from '@testing-library/react';
import mockAxios from 'axios';
import '@testing-library/jest-dom/extend-expect';

import OrderMenus from './OrderMenus';

const sampleMenus = [{
  _id: '23472937429749283',
  name: 'Menu Item 1',
  category: {
    _id: 'category001',
    name: 'Category 001',
  },
  description: 'Sample menu item 1',
  price: 7.99,
}, {
  _id: '20398420393',
  name: 'Menu Item 2',
  category: {
    _id: 'category001',
    name: 'Category 001',
  },
  description: 'Sample menu item 2',
  price: 9.99,
}, {
  _id: '0953084',
  name: 'Menu Item 3',
  category: {
    _id: 'category002',
    name: 'Category 002',
  },
  description: 'Sample menu item 3',
  price: 20.99,
}];

jest.mock('axios');

const mockSetMenu = jest.fn();

describe('<OrderMenus />', () => {

  afterEach(cleanup);

  it('should render correctly all menu items', async () => {
    const testData = {
      data: sampleMenus,
    };
    mockAxios.get.mockResolvedValueOnce(testData);

    const {
      container,
      getByTestId,
      getAllByTestId,
    } = render(
      <OrderMenus
        setMenu={mockSetMenu}
      />,
    );

    expect(getByTestId('order-menu-list')).toHaveTextContent(/Loading/);
    const menus = await waitForElement(() => getAllByTestId(/order-menu-item/));
    expect(menus).toHaveLength(sampleMenus.length);

    expect(container).toMatchSnapshot();
  });

  it('should render menu items for selected cateogry', async () => {
    const testData = {
      data: sampleMenus,
    };
    mockAxios.get.mockResolvedValueOnce(testData);

    const categoryId = 'category001';

    const {
      container,
      getByTestId,
      getAllByTestId,
    } = render(
      <OrderMenus
        setMenu={mockSetMenu}
        category={categoryId}
      />,
    );

    expect(getByTestId('order-menu-list')).toHaveTextContent(/Loading/);
    const menus = await waitForElement(() => getAllByTestId(/order-menu-item/));

    const menusForCategory = sampleMenus
      .filter(({ category: { _id: id } }) => id === categoryId);
    
    expect(menus).toHaveLength(menusForCategory.length);

    menus.forEach(m => fireEvent.click(m));
    expect(mockSetMenu).toHaveBeenCalledTimes(menus.length);

    expect(container).toMatchSnapshot();


  });
});
