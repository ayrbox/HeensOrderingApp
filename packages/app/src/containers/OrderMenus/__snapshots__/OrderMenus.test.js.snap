// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderMenus /> should render correctly 1`] = `
<div>
  <ul
    cellheight="160"
    class="MuiList-root-2 MuiList-padding-3"
    cols="3"
    data-testid="order-menu-list"
  >
    <li
      class="MuiListItem-container-7"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15 MuiListItem-alignItemsFlexStart-11 MuiListItem-secondaryAction-16"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-21"
        >
          <span
            class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
          >
            <strong>
              Menu Item 1
            </strong>
          </span>
          <span
            class="MuiTypography-root-27 MuiTypography-body1-36 Menus-itemDesc-1"
          >
            Sample menu item 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-64"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-63"
      >
        <h6
          class="MuiTypography-root-27 MuiTypography-h6-44"
        >
          £7.99
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-container-7"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15 MuiListItem-alignItemsFlexStart-11 MuiListItem-secondaryAction-16"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-21"
        >
          <span
            class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
          >
            <strong>
              Menu Item 2
            </strong>
          </span>
          <span
            class="MuiTypography-root-27 MuiTypography-body1-36 Menus-itemDesc-1"
          >
            Sample menu item 2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-64"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-63"
      >
        <h6
          class="MuiTypography-root-27 MuiTypography-h6-44"
        >
          £9.99
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-container-7"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15 MuiListItem-alignItemsFlexStart-11 MuiListItem-secondaryAction-16"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-21"
        >
          <span
            class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
          >
            <strong>
              Menu Item 3
            </strong>
          </span>
          <span
            class="MuiTypography-root-27 MuiTypography-body1-36 Menus-itemDesc-1"
          >
            Sample menu item 3
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-64"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-63"
      >
        <h6
          class="MuiTypography-root-27 MuiTypography-h6-44"
        >
          £20.99
        </h6>
      </div>
    </li>
  </ul>
</div>
`;

exports[`<OrderMenus /> should render correctly all menu items 1`] = `
<div>
  <ul
    cellheight="160"
    class="MuiList-root-2 MuiList-padding-3"
    cols="3"
    data-testid="order-menu-list"
  >
    <li
      class="MuiListItem-container-7"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15 MuiListItem-alignItemsFlexStart-11 MuiListItem-secondaryAction-16"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-21"
        >
          <span
            class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
          >
            <strong>
              Menu Item 1
            </strong>
          </span>
          <span
            class="MuiTypography-root-27 MuiTypography-body1-36 Menus-itemDesc-1"
          >
            Sample menu item 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-64"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-63"
      >
        <h6
          class="MuiTypography-root-27 MuiTypography-h6-44"
        >
          £7.99
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-container-7"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15 MuiListItem-alignItemsFlexStart-11 MuiListItem-secondaryAction-16"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-21"
        >
          <span
            class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
          >
            <strong>
              Menu Item 2
            </strong>
          </span>
          <span
            class="MuiTypography-root-27 MuiTypography-body1-36 Menus-itemDesc-1"
          >
            Sample menu item 2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-64"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-63"
      >
        <h6
          class="MuiTypography-root-27 MuiTypography-h6-44"
        >
          £9.99
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-container-7"
    >
      <div
        class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15 MuiListItem-alignItemsFlexStart-11 MuiListItem-secondaryAction-16"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-21"
        >
          <span
            class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
          >
            <strong>
              Menu Item 3
            </strong>
          </span>
          <span
            class="MuiTypography-root-27 MuiTypography-body1-36 Menus-itemDesc-1"
          >
            Sample menu item 3
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-64"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-63"
      >
        <h6
          class="MuiTypography-root-27 MuiTypography-h6-44"
        >
          £20.99
        </h6>
      </div>
    </li>
  </ul>
</div>
`;

exports[`<OrderMenus /> should render menu items for selected cateogry 1`] = `
<div>
  <ul
    cellheight="160"
    class="MuiList-root-72 MuiList-padding-73"
    cols="3"
    data-testid="order-menu-list"
  >
    <li
      class="MuiListItem-container-77"
    >
      <div
        class="MuiButtonBase-root-88 MuiListItem-root-76 MuiListItem-default-79 MuiListItem-gutters-84 MuiListItem-button-85 MuiListItem-alignItemsFlexStart-81 MuiListItem-secondaryAction-86"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-91"
        >
          <span
            class="MuiTypography-root-97 MuiTypography-subheading-104 MuiListItemText-primary-94"
          >
            <strong>
              Menu Item 1
            </strong>
          </span>
          <span
            class="MuiTypography-root-97 MuiTypography-body1-106 Menus-itemDesc-71"
          >
            Sample menu item 1
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-134"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-133"
      >
        <h6
          class="MuiTypography-root-97 MuiTypography-h6-114"
        >
          £7.99
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-container-77"
    >
      <div
        class="MuiButtonBase-root-88 MuiListItem-root-76 MuiListItem-default-79 MuiListItem-gutters-84 MuiListItem-button-85 MuiListItem-alignItemsFlexStart-81 MuiListItem-secondaryAction-86"
        data-testid="order-menu-item"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root-91"
        >
          <span
            class="MuiTypography-root-97 MuiTypography-subheading-104 MuiListItemText-primary-94"
          >
            <strong>
              Menu Item 2
            </strong>
          </span>
          <span
            class="MuiTypography-root-97 MuiTypography-body1-106 Menus-itemDesc-71"
          >
            Sample menu item 2
          </span>
        </div>
        <span
          class="MuiTouchRipple-root-134"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root-133"
      >
        <h6
          class="MuiTypography-root-97 MuiTypography-h6-114"
        >
          £9.99
        </h6>
      </div>
    </li>
  </ul>
</div>
`;
